pkg/jobs/registry.go

```diff
@@ -363,11 +364,38 @@ func (r *Registry) maybeCancelJobs(ctx context.Context, nl NodeLiveness) {
 	}
 }
 
+// isOrphaned tries to detect if there are no mutations left to be done for the
+// job which will make it a candidate for garbage collection. Jobs can be left
+// in such inconsistent state if they fail before being removed from the jobs table.
+func (r *Registry) isOrphaned(ctx context.Context, payload *jobspb.Payload) (bool, error) {
+	if payload.Type() != jobspb.TypeSchemaChange {
+		return false, nil
+	}
+	for _, id := range payload.DescriptorIDs {
+		var noMutations bool
+		if err := r.db.Txn(ctx, func(ctx context.Context, txn *client.Txn) error {
+			td, err := sqlbase.GetTableDescFromID(ctx, txn, id)
+			if err != nil {
+				return err
+			}
+			if len(td.GetMutations()) != 0 {
```

Comment:
can just assign the bool to the if condition instead of using an if, right?

Also, this seems like the name vs definition is inverted, no?